web-integration/tests/fixtures/rill-dev-fixtures.ts

```diff
@@ -0,0 +1,112 @@
+import type { Page } from "@playwright/test";
+import { asyncWaitUntil } from "@rilldata/web-common/lib/waitUtils";
+import { ADMIN_STORAGE_STATE } from "@rilldata/web-integration/tests/constants";
+import axios from "axios";
+import { spawn } from "node:child_process";
+import { cpSync, existsSync, mkdirSync, rmSync } from "node:fs";
+import { join } from "node:path";
+import { test as base } from "playwright/test";
+import treeKill from "tree-kill";
+import { getOpenPort } from "../utils/get-open-port";
+import { makeTempDir } from "../utils/make-temp-dir";
+import { spawnAndMatch } from "../utils/spawn";
+
+type MyFixtures = {
+  cliHomeDir: string;
+  project: string | undefined;
+  rillDevPage: Page;
+};
+
+export const rillDev = base.extend<MyFixtures>({
+  // Add a default home if cliHome is not provided so that tests always have a different home than the user's home.
+  // This will make sure that login status won't conflicts with dev's login status when run locally.
+  cliHomeDir: [makeTempDir("home"), { option: true }],
+  project: [undefined, { option: true }],
+
+  rillDevPage: async ({ browser, project, cliHomeDir }, use) => {
+    const TEST_PORT = await getOpenPort();
+    const TEST_GRPC_PORT = await getOpenPort();
+    const TEST_PROJECT_DIRECTORY = makeTempDir(`projects-${TEST_PORT}`);
```

Comment:
Nit: might it help to declare a test ID and use it in both temporary directories, e.g. something like:
```ts
const testId = `test-${TEST_PORT}`; // or use a random string, or test name
const cliHomeDir = makeTempDir(`cli-home-${testId}-`);
const projectDir = makeTempDir(`project-${testId}-`);
```